tests/integration/construct_component_configure_provider/go/main.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	tls "github.com/pulumi/pulumi-tls/sdk/v4/go/tls"
+	metaprovider "github.com/pulumi/pulumi/pkg/codegen/testing/test/testdata/methods-return-plain-resource/go/metaprovider"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		cfg := config.New(ctx, "")
+		proxy := cfg.Require("proxy")
+
+		configurer, err := metaprovider.NewConfigurer(ctx, "configurer", &metaprovider.ConfigurerArgs{
+			TlsProxy: unknownIfDryRun(ctx, proxy),
+		})
+		if err != nil {
+			return err
+		}
+
+		prov, err := configurer.TlsProvider(ctx)
+		if err != nil {
+			return err
+		}
+
+		key, err := tls.NewPrivateKey(ctx, "my-private-key", &tls.PrivateKeyArgs{
+			Algorithm:  pulumi.String("ECDSA"),
+			EcdsaCurve: pulumi.String("P384"),
+		}, pulumi.Provider(prov))
```

Comment:
This is the intended use case, where Configurer remote component resource preconfigured a TLS provider, and the plain-returning method extracted a reference to it, which we can use to provision other resources.